.github/workflows/codeql-query.yml

```diff
@@ -96,7 +96,7 @@ jobs:
 
   combine-results:
     runs-on: ubuntu-latest
-    if: always()
+    if: ${{ always() && github.event.inputs.variant_analysis_id == '' }}
```

Comment:
The `${{` shouldn't be necessary according to the [expressions page](https://docs.github.com/en/actions/learn-github-actions/expressions).

Have you tested whether this will evaluate correctly? For the `post-if` expression inside an `action.yml` file, this would actually not work because it's evaluating an expression inside an implicit expression and the `failure` function was undefined.

Another question: Are we able to use [the `inputs` context](https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context) instead of `github.event.inputs`?

Reply:
I've removed the `${{`. Thanks for pointing that out.

I don't know about the `inputs` context but I expect we just didn't know about it or didn't think too hard about it since `github.event.inputs` worked. I can see that it's a bit shorter though and that's nice. Just to keep changes separate I'll make that change in a followup PR.

I've also now tested this and it seems to be working correctly. Here are a few different runs:
- A run in live results mode - https://github.com/dsp-testing/qc-controller/actions/runs/3104378048
- A run in the old artifacts mode - https://github.com/dsp-testing/qc-controller/actions/runs/3104408456
- A run in the old artifacts mode where I cancelled it half way - https://github.com/dsp-testing/qc-controller/actions/runs/3104420815